python/hsfs/engine/python.py

```diff
@@ -963,36 +963,40 @@ def acked(err, msg):
             producer.flush()
             progress_bar.close()
 
-        # start backfilling job
-        # if topic didn't exist, always run the backfill job to reset the offsets except if it's a multi insert
+        # start materialization job
+        # if topic didn't exist, always run the materialization job to reset the offsets except if it's a multi insert
         if (
             not isinstance(feature_group, ExternalFeatureGroup)
             and reset_offsets
             and not feature_group._multi_part_insert
         ):
-            if offline_write_options is not None and not offline_write_options.get(
-                "start_offline_backfill", True
+            if offline_write_options is not None and not (
+                offline_write_options.get("start_offline_materialization", True)
+                or offline_write_options.get("start_offline_backfill", True)
```

Comment:
This if statement doesn't work. Users will only set one of the two (Let's say `start_offline_materialization` to `False`). The other one is going to be empty and so the results is going to be True.

I think the easiest is to remove the default value and check if the keys exists.